### PR TITLE
Geodetic training loop

### PR DESCRIPTION
**Pull request by albangossard**
Includes many changes, sorry for the massive PR!
Changes include:
- Add automatic computation of sky view factor in a dedicated venv #15 
- Add automatic creation of gridded features for #112
- Add support of dhdt from Hugonnet et al 2021 #16 
- Add Iceland data download + preprocessing #170 
- Fix projection to retrieve OGGM features #78 
- Implement parallel async geodetic data loading #110 
- Moved some functions to fix circular dependencies
- Changed how normalization is performed to improve efficiency
- Improved how the padding is computed to be more robust and support measurements spanning up to 3 hydrological years
- Added profile and cumulated MB plots
- Added Sinkhorn distance optimization to create the train/test sets which are close in the distribution sense
- Currently adding profiling in the geodetic training loop, this is still a work in progress

### CODE REVIEW

File: massbalancemachine/data_processing/oggm_utils.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        oggmCfg.PATHS["working_dir"] = custom_working_dir
+
+
+def _initialize_glacier_directories(rgi_ids_list: list, cfg: config.Config) -> list:
+    """Initialize glacier directories."""
+    base_url = cfg.base_url_w5e5
+    glacier_directories = workflow.init_glacier_directories(
+        rgi_ids_list,
+        reset=False,
+        from_prepro_level=3,
```
Comment (marvande) [inline massbalancemachine/data_processing/oggm_utils.py]: do we need prepro_level 3 ? I've been running it at 2 since level 3 gives the climate features and the ice flow inversions but we don't need those. Using 2 might speed things up ?

Comment (marvande) [reply massbalancemachine/data_processing/oggm_utils.py]: And I think level 3 is not available for all RGIs if I remember well. Actually we might also just need level 1 ? 

From the doc: 

Level 1: directories now contain the glacier topography data as well.

Level 2: at this stage, the glacier flowlines and their downstream lines are computed and ready to be used.

Level 3: has the baseline climate timeseries (e.g. W5E5, CRU or ERA5) added to the directories. It also contains all necessary pre-processing tasks for a dynamical run, including the mass balance calibration, bed inversion, up to the [tasks.init_present_time_glacier()](https://docs.oggm.org/en/stable/generated/oggm.tasks.init_present_time_glacier.html#oggm.tasks.init_present_time_glacier) task. These directories still contain all data that were necessary for the processing. Therefore they are large in size but also the most flexible since the processing chain can be re-run from them.

**Review by marvande — COMMENTED** [review]
Seems good, just a small question for the OGGM retrieval :)

**Comment by marvande** [comment]
Also, could you please add the changes in #183 made to [docs/install.rst‎](https://github.com/ODINN-SciML/MassBalanceMachine/pull/183/changes#diff-379774f5c6e537f2001c951863c6ac577b56eb5c6e61cc581dce95143760ff36) ? It's an issue I ran into when trying to remake the environment and it's probably good if the fix is in there as soon as possible :)

File: massbalancemachine/data_processing/oggm_utils.py (hunk 1 above)
Comment (albangossard) [reply massbalancemachine/data_processing/oggm_utils.py]: Good idea thanks! I moved the code without really checking what it was doing. I changed the default preprocessing level to 2 in 19ce035a03b7fa390b20db127c20ef40bc3ce2eb

**Review by marvande — APPROVED** [review]
